<p>
  <code>if</code> statements with conditions that are always false have the effect of making blocks of code non-functional. <code>if</code> statements with conditions that are always true are completely redundant, and make the code less readable.

  <p>
  There are three possible causes for the presence of such code:
  </p>
  <ul>
    <li>An if statement was changed during debugging and that debug code has been committed.</li>
    <li>Some value was left unset.</li>
    <li>Some logic is not doing what the programmer thought it did.</li>
  </ul>

</p>
<p>
In any of these cases, unconditional <code>if</code> statements should be removed.
</p>


<h2>Noncompliant Code Example</h2>
<pre>
if (true) {
  doSomething();
}

if (false) {
  doSomethingElse();
}
</pre>

<h2>Compliant Solution</h2>
<pre>
doSomething();
</pre>
