<p>Unused parameters are misleading. Whatever the value passed to such parameters is, the behavior will be the same.</p>
<h2>Noncompliant Code Example</h2>
<pre>
function doSomething($a, $b) { // "$a" is unused
  return compute($b);
}
</pre>
<h2>Compliant Solution</h2>
<pre>
function doSomething($b) {
  return compute($b);
}
</pre>
<h2>Exceptions</h2>
<p>Functions in classes that override a class or implement interfaces are ignored.</p>
<pre>
class C extends B {

  function doSomething($a, $b) {     // no issue reported on $b
    compute($a);
  }

}
</pre>
<h2>See</h2>
<ul>
  <li> MISRA C++:2008, 0-1-11 - There shall be no unused parameters (named or unnamed) in nonvirtual functions. </li>
  <li> MISRA C:2012, 2.7 - There should be no unused parameters in functions </li>
  <li> <a href="https://www.securecoding.cert.org/confluence/x/NYA5">CERT, MSC12-C.</a> - Detect and remove code that has no effect or is never
  executed </li>
</ul>

