<h2>Why is this an issue?</h2>
<p>Unused parameters are misleading. Whatever the value passed to such parameters is, the behavior will be the same.</p>
<h3>Noncompliant code example</h3>
<pre>
function doSomething($a, $b) { // "$a" is unused
  return compute($b);
}
</pre>
<h3>Compliant solution</h3>
<pre>
function doSomething($b) {
  return compute($b);
}
</pre>
<h3>Exceptions</h3>
<p>Functions in classes that override a class or implement interfaces are ignored.</p>
<pre>
class C extends B {

  function doSomething($a, $b) {     // no issue reported on $b
    compute($a);
  }

}
</pre>

