<h2>Why is this an issue?</h2>
<p>A typical code smell known as unused function parameters refers to parameters declared in a function but not used anywhere within the function’s
body. While this might seem harmless at first glance, it can lead to confusion and potential errors in your code. Disregarding the values passed to
such parameters, the function’s behavior will be the same, but the programmer’s intention won’t be clearly expressed anymore. Therefore, removing
function parameters that are not being utilized is considered best practice.</p>
<h3>Exceptions</h3>
<p>Functions in classes that override a class or implement interfaces are ignored.</p>
<pre>
class C extends B {

  function doSomething($a, $b) {     // no issue reported on $b
    compute($a);
  }
}
</pre>
<p>Additionally, methods with specific Magento plugin signatures are ignored because Magento requires these parameters:</p>
<ul>
  <li> Methods starting with <code>before</code> and having at least 3 parameters </li>
  <li> Methods starting with <code>around</code> and having at least 4 parameters </li>
  <li> Methods starting with <code>after</code> and having at least 2 parameters </li>
</ul>
<pre>
class ExamplePlugin {

  public function beforeAction($subject, $arg1, $arg2) {
    // no issue on unused parameters, required by Magento plugin signature
    return;
  }

  public function aroundAction($subject, callable $proceed, $arg1, $arg2) {
    // no issue on unused parameters, required by Magento plugin signature
    return;
  }

  public function afterAction($subject, $result) {
    // no issue on unused parameters, required by Magento plugin signature
    return;
  }
}
</pre>
<h2>How to fix it</h2>
<p>Having unused function parameters in your code can lead to confusion and misunderstanding of a developer’s intention. They reduce code readability
and introduce the potential for errors. To avoid these problems, developers should remove unused parameters from function declarations.</p>
<h3>Code examples</h3>
<h4>Noncompliant code example</h4>
<pre data-diff-id="1" data-diff-type="noncompliant">
function doSomething($a, $b) { // "$a" is unused
  return compute($b);
}
</pre>
<h4>Compliant solution</h4>
<pre data-diff-id="1" data-diff-type="compliant">
function doSomething($b) {
  return compute($b);
}
</pre>

