<p>When the call to a function doesn't have any side effect, what is the point of making the call if the results are ignored? In such cases, either
the function call is useless and should be dropped, or the source code doesn't behave as expected. </p>
<h2>Noncompliant Code Example</h2>
<pre>
strlen($name); // Noncompliant; "strlen" has no side effect
</pre>
<h2>Compliant Solution</h2>
<pre>
$length = strlen($name);
</pre>
<h2>See</h2>
<ul>
  <li> MISRA C:2012, 17.7 - The value returned by a function having non-void return type shall be used </li>
  <li> <a href="https://www.securecoding.cert.org/confluence/x/9YIRAQ">CERT, EXP12-C.</a> - Do not ignore values returned by functions </li>
  <li> <a href="https://www.securecoding.cert.org/confluence/x/9gEqAQ">CERT, EXP00-J.</a> - Do not ignore values returned by methods </li>
</ul>

