<h2>Why is this an issue?</h2>
<p>WordPress has a database repair and optimization mode that can be activated by setting <code>WP_ALLOW_REPAIR</code> to <code>true</code> in the
configuration.</p>
<p>If activated, the repair page can be accessed by any user, authenticated or not. This makes sense because if the database is corrupted, the
authentication mechanism might not work.</p>
<p>Malicious users could trigger this potentially costly operation repeatadly slowing down the website, and making it unavailable.</p>
<h2>How to fix it</h2>
<p>Itâ€™s recommended to enable automatic database repair mode only in case of database corruption. This feature should be deactivated again when the
database issue is resolved.</p>
<h3>Code Examples</h3>
<h4>Noncompliant Code Example</h4>
<pre data-diff-id="1" data-diff-type="noncompliant">
define( 'WP_ALLOW_REPAIR', true ); // Noncompliant
</pre>
<h4>Compliant Solution</h4>
<pre data-diff-id="1" data-diff-type="compliant">
// The default value is false, so the value does not have to be expilicitly set.
define( 'WP_ALLOW_REPAIR', false );
</pre>
<h2>Resources</h2>
<ul>
  <li> OWASP - <a href="https://owasp.org/Top10/A05_2021-Security_Misconfiguration/">Top 10 2021 Category A5 - Security Misconfiguration</a> </li>
  <li> OWASP - <a href="https://owasp.org/Top10/A07_2021-Identification_and_Authentication_Failures/">Top 10 2021 Category A7 - Identification and
  Authentication Failures</a> </li>
  <li> <a href="https://wordpress.org/support/article/editing-wp-config-php/#automatic-database-optimizing">wordpress.org</a> - Automatic Database
  Optimizing </li>
  <li> OWASP - <a href="https://owasp.org/www-project-top-ten/2017/A6_2017-Security_Misconfiguration">Top 10 2017 Category A6 - Security
  Misconfiguration</a> </li>
</ul>

